Capitalizing City Name Before Save - in progress

dondonz opened the pull request:
This Addresses Issue #535.

In The Group Model, I've Added A `before_save` Method To Capitalize A City's Name Before Saving To The Database.

dondonz commented:
Next up - update the method for cities with 2 word names (eg New York)

til commented:
Cool, thanks for starting with this!

One minor style remark - the extra empty lines before the end of a class are not necessary. The ruby style guide says they should not be added: https://github.com/bbatsov/ruby-style-guide#empty-lines-around-bodies - I'm sure though we are totally not consistent with this, and have a mixture of all styles in our current codebase.

dondonz commented:
@til Thanks! I've removed the line breaks before the end.

I've changed .capitalize to .titleize. If a user types 'berlin' or 'new york' it will now be converted to 'Berlin' or 'New York'.

I aimed to solve the case where a user accidentally types in the lower case form of the name. Of course, there are infinite possibilities for misspelling:

- If a user types 'new yoRk', the name will become 'New Yo Rk'.
- If a user types 'Berlin2', the name will still be 'Berlin2'

And... another edge case:
- If a user lives in a town with a hyphenated name, 'Place-Name' will become 'Place Name'

What do you think - shall we stop at .titleize?

til commented:
`titleize` - good find. There seems to be a method for everything in the
Rails code.

I think this solution is great. Looking at the list of cities we have
currently, there is nothing exotic in it. One day we might get a problem
when someone enters a city with a dash (do those even exist?), but we
might look into that once it happens, if ever.

Do you want to squash and rebase onto master?

dondonz commented:
Fantastic! Thanks @til!

lislis commented:
Awesome!
I cleaned up the data on the production database. I'll merge this and deploy!

lislis commented:
AAAAAnd deployed!
![wheee](https://media.giphy.com/media/3og0IT9dAZyMz3lXNe/200w_d.gif)

til commented:
Great!
